### ISSUE TITLE
Slide type UI

### ISSUE THREAD

**Issue by pwsiegel** [jupyterlab/jupyterlab#7460]
I just discovered the jupyterlab feature which allows one to export a notebook as a reveal.js presentation, and this is absolutely amazing.  I'm almost ready to go hog wild and start using this as my standard tool for writing presentations, but there are a couple UI issues that give me pause:

- There do not appear to be keyboard shortcuts for changing the slide type
- It is not possible to determine the slide type of a cell without selecting it

Are there any workarounds for these?  Does anyone with experience making Jupyter extensions have intuition for how hard it would be to add these features as an extension?

**Comment by jasongrout** [comment]
> Are there any workarounds for these?

Not right now.

> It is not possible to determine the slide type of a cell without selecting it

There is other discussion about tags not being visible without selecting the cell. This won't be trivial to fix right now. Basically, we'd need to agree on how to change the ux to indicate cell tags inline with the cells.

> There do not appear to be keyboard shortcuts for changing the slide type

This should be fairly straightforward to do in an extension, i.e., make a new command that gets the current cell and adds or changes the cell metadata. Then bind a keyboard shortcut to that command.

We should have a command for changing cell metadata in core, I think. Then it would be easy to make a keyboard shortcut for invoking that command.

**Comment by jasongrout** [comment]
> There is other discussion about tags not being visible without selecting the cell.

This is at https://github.com/jupyterlab/jupyterlab/issues/7449

**Comment by pwsiegel** [comment]
Thank you for your response!  I might have a go at adding keyboard shortcuts via an extension - that alone would go a long way.

As I have continued to play around with the "export as reveal.js" feature, I've noticed a few other bugs / nice-to-have requests.  Before I raise any additional issues, is there already any work underway to improve this part of the system?

**Comment by jasongrout** [comment]
I don't know of anyone working on improving the slides per se, but as you see from the cell-tags extension and discussion, people are working on improving tagging cells.

**Comment by jasongrout** [comment]
> We should have a command for changing cell metadata in core, I think.

I think the way I would (at least start) doing this is making a command that gets the selected cells in the current notebook and takes as args the metadata to add/delete/change, and does the necessary changes.

Then you can make a keyboard shortcut with the command and appropriate args to change the cell metadata as you wish.

**Comment by sglyon** [comment]
I went ahead and did this here: https://www.npmjs.com/package/jupyterlab-slide-mode-keyboard-shortcuts


@palewire please check it out and let me know how it works for you

**Comment by flying-sheep** [comment]
what’s the state of this 5½ years later?

**Comment by krassowski** [comment]
I know that these two extensions are maintained:
- https://github.com/jupyterlab-contrib/jupyterlab-slideshow
- https://github.com/jupyterlab-contrib/rise

Do they address the original issue?

**Comment by flying-sheep** [comment]
yes, but they are a bit hard to find, since jupyterlab-deck isn’t maintained, and RISE gives a lot of old results for notebook only.
